### PR TITLE
remove deprecated items

### LINKED ISSUE #3793: Advance depcreation markers for 2.1 release
Things with `DeprecatedIn19` can be removed now... or become `PersistentlyDeprecated`, whichever:

- [ ] `MACContext`
- [ ] `UnsupportedExtension`
- [ ] `RIPEMD160`
- [ ] `Whirlpool`

### PR DESCRIPTION
**Pull request by reaperhulk**
whirlpool, ripemd160, unsupportedextension, and the old interfaces

followup PR will remove RIPEMD160 and Whirlpool test vectors.

fixes #3793

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,6 +6,9 @@ Changelog
 
 .. note:: This version is not yet released and is under active development.
 
+* **BACKWARDS INCOMPATIBLE:** ``Whirlpool``, ``RIPEMD160``, and
+  ``UnsupportedExtension`` have been removed.
```
Comment (alex) [inline CHANGELOG.rst]: `MACContext` as well?

This should be clear they were already deprecated.

**Comment by alex** [comment]
There's also some unused imports, and `generate_long_string_hash_test` appears to be unused now.

File: CHANGELOG.rst (hunk 1 above)
Comment (reaperhulk) [inline CHANGELOG.rst]: We haven't removed `MACContext`, just moved it. Not sure how to phrase that.

Comment (alex) [inline CHANGELOG.rst]: Oh, right. Uhh, yeah, that's fine.

Comment (alex) [inline CHANGELOG.rst]: The bit about making clear they were deprecated still stands.

**Comment by alex** [comment]
I don't understand what's going on with docs on macOS?

**Comment by alex** [comment]
Ok, I wonder if this a segfault in pyenchant's wheels because x86 instructions or something.

**Pull request merged**